client/js/lounge.js

```diff
@@ -1310,10 +1314,58 @@ $(function() {
 						order: order
 					}
 				);
+
+				ignoreSortSync = true;
 			}
 		});
 	}
 
+	socket.on("sync_sort", function(data) {
+		// Syncs the order of channels or networks when they are reordered
+
+		if (ignoreSortSync) {
+			ignoreSortSync = false;
+			return; // Ignore syncing because we 'caused' it
+		}
+
+		var type = data.type;
+		var order = data.order;
+
+		if (type === "networks") {
+			var container = $(".networks");
+
+			$.each(order, function(key, value) {
+				var position = $(container.children()[key]);
+
+				if (position.data("id") === value) { // Network in correct place
+					return true; // No point in continuing
+				}
+
+				var network = container.find("#network-" + value);
+
+				$(network).insertBefore(position);
+			});
+		} else if (type === "channels") {
+			var network = $("#network-" + data.target);
+
+			$.each(order, function(key, value) {
+				if (key === 0) { // Shouldn't attempt to move lobby
+					return true; // `continue` -> skip to next item
+				}
+
+				var position = $(network.children()[key]); // Target channel at position
+
+				if (position.data("id") === value) { // Channel in correct place
+					return true; // No point in continuing
+				}
+
+				var channel = network.find(".chan[data-id=" + value + "]"); // Channel at position
+
+				$(channel).insertBefore(position);
```

Comment:
Rearranging the channels currently does not update the id's, not even after refreshing the page so I didn't think of doing it. It seems like a server side thing, should I add it?